src/rigid_filament.cpp

```diff
@@ -161,13 +164,15 @@ void RigidFilament::Integrate() {
    with std dev sqrt(2*kT*dt/gamma) where gamma is the friction
    coefficient along that direction */
 void RigidFilament::AddRandomDisplacement() {
+  double mag = 0;
   // Get vector(s) orthogonal to orientation
   GetBodyFrame();
   // First handle the parallel component
-  double mag = rng_.RandomNormal(diffusion_par_);
-  for (int i = 0; i < n_dim_; ++i)
-    position_[i] += mag * orientation_[i];
-  // Then the perpendicular component(s)
+  if (!pseudo1D_) {
+    mag = rng_.RandomNormal(diffusion_par_);
+    for (int i = 0; i < n_dim_; ++i) position_[i] += mag * orientation_[i];
```

Comment:
Make sure to follow code conventions. Run clang-format on code before submitting review. https://www.electronjs.org/docs/development/clang-format#:~:text=clang%2Dformat%20is%20a%20tool,you%20and%20the%20reviewers'%20time.